### PR TITLE
feat: migrate scalar responses to new enum pattern

### PR DESCRIPTION
**Pull request by anitarua**
Closes https://github.com/momentohq/dev-eco-issue-tracker/issues/844

### CODE REVIEW

File: examples/nodejs/cache/advanced.ts [hunk 1]
```diff
@@ -96,8 +96,6 @@ async function setGetDeleteExample() {
     logger.info(`Error deleting cache key: ${deleteResponse.message()}`);
   } else if (deleteResponse instanceof CacheDelete.Success) {
     logger.info('Deleted key from cache');
-  } else {
```
Comment (bruuuuuuuce) [inline examples/nodejs/cache/advanced.ts:99]: :w00t:

File: packages/client-sdk-web/src/internal/cache-data-client.ts [hunk 2]
```diff
@@ -431,7 +431,7 @@ export class CacheDataClient<
                 break;
               default:
                 resolve(
-                  new CacheGet.Error(
+                  new CacheSetIfNotExists.Error(
```
Comment (bruuuuuuuce) [inline packages/client-sdk-web/src/internal/cache-data-client.ts:434]: was this a bug in the old code?

Comment (anitarua) [reply packages/client-sdk-web/src/internal/cache-data-client.ts:434]: yup

Comment (bruuuuuuuce) [reply packages/client-sdk-web/src/internal/cache-data-client.ts:434]: :w00t:

**Pull request merged**